test/integration/zap-mint.ts

```diff
@@ -0,0 +1,239 @@
+import { EXTERNAL_USD_STABLE } from "./../../utils/Constants";
+import { BigNumber } from "ethers";
+import hre from "hardhat";
+import { ZapMint } from "./../../typechain/ZapMint";
+import chai from "chai";
+import { solidity } from "ethereum-waffle";
+import cap from "chai-as-promised";
+import {
+  getAllBDStablePools,
+  getBdEu,
+  getBDStableWethPool,
+  getBdUs,
+  getBdx,
+  getDeployer,
+  getERC20,
+  getTreasurySigner,
+  getUniswapRouter,
+  getWbtc,
+  getWeth
+} from "../../utils/DeployedContractsHelpers";
+import { numberToBigNumberFixed, to_d18 } from "../../utils/NumbersHelpers";
+import { setUpFunctionalSystemForTests } from "../../utils/SystemSetup";
+import { generateAllPaths, getAllTokensAddresses } from "../../utils/UniswapPoolsHelpers";
+import { IERC20 } from "../../typechain/IERC20";
+import { BdStablePool } from "../../typechain/BdStablePool";
+import { expectToFail } from "../helpers/common";
+
+chai.use(cap);
+
+chai.use(solidity);
+const { expect } = chai;
+
+//todo
+//add test for collateral ratio < 100% (mint which use bdx)
+//use one initialized zapmint object for all tests..? (alternatively - move initialization to private method)
+//move common variables to constants (like deadline, router)
+//add test for zap minting directly from ETH/WBTC (ETH/WBTC in path[0], path.length = 1)
+//add tests for all require statements
+//add test for reentrancy?
+describe("Zap mint", () => {
```

Comment:
we should add tests which include multiple routers

Reply:
yep, all the tasks to do regarding support for multiple routers will be in https://lagoslabs.atlassian.net/browse/LAGO-901